src/neural/wrapper.cc

```diff
@@ -0,0 +1,148 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2024 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+
+  Additional permission under GNU GPL version 3 section 7
+
+  If you modify this Program, or any covered work, by linking or
+  combining it with NVIDIA Corporation's libraries from the NVIDIA CUDA
+  Toolkit and the NVIDIA CUDA Deep Neural Network library (or a
+  modified version of those libraries), containing parts covered by the
+  terms of the respective license agreement, the licensors of this
+  Program grant you additional permission to convey the resulting work.
+*/
+
+#include "neural/wrapper.h"
+
+#include <algorithm>
+#include <numeric>
+
+#include "neural/encoder.h"
+#include "utils/fastmath.h"
+
+namespace lczero {
+namespace {
+
+class NetworkAsBackend : public Backend {
+ public:
+  NetworkAsBackend(std::unique_ptr<Network> network,
+                   float softmax_policy_temperature)
+      : network_(std::move(network)),
+        softmax_policy_temperature_(1.0f / softmax_policy_temperature) {
+    const NetworkCapabilities& caps = network_->GetCapabilities();
+    attrs_.has_mlh = caps.has_mlh();
+    attrs_.has_wdl = caps.has_wdl();
+    attrs_.runs_on_cpu = network_->IsCpu();
+    attrs_.suggested_num_search_threads = network_->GetThreads();
+    attrs_.recommended_batch_size = network_->GetMiniBatchSize();
+    attrs_.maximum_batch_size = std::numeric_limits<int>::max();
```

Comment:
A safe value to use for `maximum_batch_size` is 1024, as this is the maximum batch size lc0 will make.

Reply:
Updated. In general it's not nice to hardcode it like that, but given wrapper is only for older backends, ok.

Reply:
This should definitely be provided by the backend even for old ones, but 1024 will do until we implement a mechanism for it.